lib/Net/Amazon/S3/Client/Object.pm

```diff
@@ -263,11 +263,13 @@ sub delete {
 
 sub initiate_multipart_upload {
     my $self = shift;
+    my %args = ref($_[0]) ? %{$_[0]} : @_;
     my $http_request = Net::Amazon::S3::Request::InitiateMultipartUpload->new(
         s3     => $self->client->s3,
         bucket => $self->bucket->name,
         key    => $self->key,
         encryption => $self->encryption,
+        %args,
```

Comment:
I think safer and easier to understand will be
```suggestion
        (headers => $args{headers}) x!! exists $args{headers},
```

Reply:
That's what i was doing initially :) thanks for the feedback